scripts/ci/pypi_nightly_tag.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+"""
+Idea from https://github.com/streamlit/streamlit/blob/4841cf91f1c820a392441092390c4c04907f9944/scripts/pypi_nightly_create_tag.py
+"""
+
+from datetime import datetime
+import sys
+
+import packaging.version
+import pytz
+from packaging.version import Version
+
+PYPI_LANGFLOW_URL = "https://pypi.org/pypi/langflow/json"
+PYPI_LANGFLOW_NIGHTLY_URL = "https://pypi.org/pypi/langflow-nightly/json"
+
+PYPI_LANGFLOW_BASE_URL = "https://pypi.org/pypi/langflow-base/json"
+PYPI_LANGFLOW_BASE_NIGHTLY_URL = "https://pypi.org/pypi/langflow-base-nightly/json"
+
+
+def get_latest_published_version(build_type: str, is_nightly: bool) -> Version:
+    import requests
+
+    url = ""
+    if build_type == "base":
+        url = PYPI_LANGFLOW_BASE_NIGHTLY_URL if is_nightly else PYPI_LANGFLOW_BASE_URL
+    elif build_type == "main":
+        url = PYPI_LANGFLOW_NIGHTLY_URL if is_nightly else PYPI_LANGFLOW_URL
+    else:
+        raise ValueError(f"Invalid build type: {build_type}")
+
+    res = requests.get(url)
+    try:
+        version_str = res.json()["info"]["version"]
+    except Exception as e:
+        raise RuntimeError("Got unexpected response from PyPI", e)
+    return Version(version_str)
+
+
+def create_tag(build_type: str):
+    current_version = get_latest_published_version(build_type, is_nightly=False)
+    current_nightly_version = get_latest_published_version(build_type, is_nightly=True)
+
+    build_number = "0"
+    latest_base_version = current_version.base_version
+    nightly_base_version = current_nightly_version.base_version
+
+    if latest_base_version == nightly_base_version:
+        # If the latest version is the same as the nightly version, increment the build number
+        build_number = str(current_nightly_version.dev + 1)
+
+    new_nightly_version = latest_base_version + ".dev" + build_number
+
+    # X.Y.Z.dev.YYYYMMDD
+    # This takes the base version of the current version and appends the
+    # current date. If the last release was on the same day, we exit, as
+    # pypi does not allow for overwriting the same version.
+    #
+    # We could use a different versioning scheme, such as just incrementing
+    # an integer.
+    # version_with_date = (
+    #     ".".join([str(x) for x in current_version.release])
+    #     + ".dev"
+    #     + "0"
+    #     + datetime.now(pytz.timezone("UTC")).strftime("%Y%m%d")
+    # )
```

Comment:
Note this. The current versioning uses `X.Y.Z.devN` - this is a lot easier to work with when iterating, as you can just increment `N`, but I see the appeal of the date-based name once builds are stable here. We can push this for now, ensure it's working in the next few iterations, yank the `.devN` builds, then move to date-based.

Reply:
I like the date-based as well and agree that we can change that later